config/initializers/sidekiq.rb

```diff
@@ -32,10 +32,13 @@
 
 Sidekiq.configure_server do |config|
   config.on(:startup) do
-    Sidekiq::Cron::Job.load_from_hash YAML.load_file("config/sidekiq_cron_schedule.yml")
+    unless FeatureToggle.maintenance_mode_enabled?
+      Sidekiq::Cron::Job.load_from_hash YAML.load_file("config/sidekiq_cron_schedule.yml")
+    end
   end
 
   config.on(:shutdown) do
     Sidekiq::CLI.instance.launcher.quiet
+    Sidekiq::Cron::Job.all.each(&:destroy)
```

Comment:
Without this line, jobs will not be destroyed during a deployment. So enabling maintenance mode wouldn't turn off the data export xml job. Also on staging right now there is still the data export csv job running (and failing) every day - it would be good to clean bits like that up.

We could probably also destroy the jobs in the startup block, but I don't think it matters. Also we could specify specific jobs to destroy instead of all of them, but it seems wise to me to destroy everything!

Reply:
Note that AFACIT there were no tests for this sidekiq config/job stuff. Seems wise for me to write some, shout if you think that's not worth the time right now though.

Reply:
Does this apply every time an app instance is shut down?

Reply:
As agreed, we want to move this line into the server startup section. There's a chance that if a server starts the moment before the job is supposed to run then the job will get deleted and not recreated in time so the job won't get run. We think this is a tiny risk and is easily outweighed by the advantages of ensuring the jobs which exists are aligned with what's defined in the code.